Gale Databases.js

```diff
@@ -1,190 +1,211 @@
+	'target': '^https?://[^?&]*(?:gale|galegroup|galetesting|ggtest)\.com(?:\:\d+)?/ps/',
+	'minVersion': '3.0',
+	'maxVersion': '',
+	'priority': 100,
+	'inRepository': true,
+	'translatorType': 4,
+	'browserSupport': 'gcsibv',
+	'lastUpdated': '2019-10-11 12:00:00'
 }
 
 /*
 	***** BEGIN LICENSE BLOCK *****
-	
-	Galegroup Translator - Copyright © 2018 Sebastian Karcher 
+
+	Gale Databases Translator - Copyright © 2019
 	This file is part of Zotero.
-	
+
 	Zotero is free software: you can redistribute it and/or modify
 	it under the terms of the GNU Affero General Public License as published by
 	the Free Software Foundation, either version 3 of the License, or
 	(at your option) any later version.
-	
+
 	Zotero is distributed in the hope that it will be useful,
 	but WITHOUT ANY WARRANTY; without even the implied warranty of
 	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 	GNU Affero General Public License for more details.
-	
+
 	You should have received a copy of the GNU Affero General Public License
 	along with Zotero.  If not, see <http://www.gnu.org/licenses/>.
-	
+
 	***** END LICENSE BLOCK *****
 */
-function attr(docOrElem,selector,attr,index){var elem=index?docOrElem.querySelectorAll(selector).item(index):docOrElem.querySelector(selector);return elem?elem.getAttribute(attr):null}function text(docOrElem,selector,index){var elem=index?docOrElem.querySelectorAll(selector).item(index):docOrElem.querySelector(selector);return elem?elem.textContent:null}
 
+var GaleZotero = (function () {
+var DATA_TRANSLATOR = '32d59d2d-b65a-4da4-b0a3-bdd3cfb979e7';
+var POST_URL = '/ps/citationtools/rest/cite/download';
 
-function getSearchResults(doc, checkOnly) {
-	var items = {};
-	var found = false;
+function detect(doc) {
+	var entries = getCitableDocuments(doc);
+	switch (entries.length) {
+		case 0: return false;
+		case 1: return entries[0].getAttribute('data-zoterolabel');
+		default: return 'multiple';
+	}
+}
 
-	var rows = doc.querySelectorAll('ul.SearchResultsList span.title a.documentLink');
-	if (!rows.length) {
-		rows = doc.querySelectorAll('ul.SearchResultsList p.subTitle a.title');
+function process(doc) {
+	var entries = getCitableDocuments(doc);
+	switch (entries.length) {
+		case 0: break;
+		case 1: processSingleEntry(doc); break;
+		default: processMulipleEntries(entries);
 	}
-	for (var i=0; i<rows.length; i++) {
-		var href = rows[i].href;
-		var title = ZU.trimInternal(rows[i].textContent);
-		if (!href || !title) continue;
-		if (checkOnly) return true;
-		found = true;
-		items[href] = title;
+}
+
+function processMulipleEntries(entries) {
+	Zotero.selectItems(createKeyValuePairs(entries), function (selectedItems) {
+		if (selectedItems) {
+			Zotero.Utilities.processDocuments(getURLs(selectedItems), processSingleEntry);
+		}
+	});
+}
+
+function createKeyValuePairs(entries) {
+	var map = {};
+	for (var item in entries) {
+		/* istanbul ignore next */
+		if (entries.hasOwnProperty(item)) {
```

Comment:
This check isn't necessary — Object.prototype will never be overridden in the contexts where this will run.